lib/models/addon.js

```diff
@@ -797,16 +844,28 @@ var Addon = CoreObject.extend({
   compileAddon: function(tree) {
     this._requireBuildPackages();
 
-    var reexported;
+    var emberCLIBabelConfigKey = this._emberCLIBabelConfigKey();
+    if (!this.options[emberCLIBabelConfigKey].compileModules) {
+      throw new SilentError(
+        'Ember CLI addons manage their own module transpilation during the `treeForAddon` processing. ' +
+          '`' + this.name + '` has overridden the `this.options.' + emberCLIBabelConfigKey + '.compileModules` ' +
```

Comment:
as there may be more then one addon by this name in node_modules, should we print out its whole path?